storage/Devices/LvmVgImpl.cc

```diff
@@ -520,4 +520,102 @@ namespace storage
 	return lhs->get_vg_name() < rhs->get_vg_name();
     }
 
+
+    void
+    LvmVg::Impl::add_dependencies(Actiongraph::Impl& actiongraph) const
+    {
+	// First, all the operations removing or shrinking LVs
+	vector<Actiongraph::Impl::vertex_descriptor> decrease_lv_actions;
+	// Then a sync point
+	Action::Base* decrease_lv_sync = new Action::Create(get_sid(), true);
+	// Afterward, all the operations adding or removing PVs...
+	vector<Actiongraph::Impl::vertex_descriptor> reallot_actions;
+	// ..and the corresponding sync point
+	Action::Base* reallot_sync = new Action::Create(get_sid(), true);
+	// Finally, operations adding and growing LVs
+	vector<Actiongraph::Impl::vertex_descriptor> increase_lv_actions;
+
+	// Look for the relevant actions
+	for (Actiongraph::Impl::vertex_descriptor vertex : actiongraph.vertices())
+	{
+	    const Action::Base* action = actiongraph[vertex];
+
+	    if (action_is_my_reallot(action, actiongraph))
+		reallot_actions.push_back(vertex);
+	    else if (action_is_on_my_lv(action, actiongraph))
+	    {
+		if (action_frees_vg_space(action))
+		    decrease_lv_actions.push_back(vertex);
+		else
+		    increase_lv_actions.push_back(vertex);
+	    }
+	}
+
+	// Add the dependencies to the action graph
+
+	Actiongraph::Impl::vertex_descriptor sync1 = actiongraph.add_vertex(decrease_lv_sync);
+	for (const Actiongraph::Impl::vertex_descriptor& vertex : decrease_lv_actions)
+	    actiongraph.add_edge(vertex, sync1);
+
+	Actiongraph::Impl::vertex_descriptor sync2 = actiongraph.add_vertex(reallot_sync);
+	for (const Actiongraph::Impl::vertex_descriptor& vertex : reallot_actions)
+	{
+	    actiongraph.add_edge(sync1, vertex);
+	    actiongraph.add_edge(vertex, sync2);
+	}
+
+	for (const Actiongraph::Impl::vertex_descriptor& vertex : increase_lv_actions)
+	    actiongraph.add_edge(sync2, vertex);
```

Comment:
I don't think it will work. I will create a test case to make sure we don't skip that case.